pkg/observer/controller.go

```diff
@@ -272,7 +272,8 @@ func (c *controller) safeToStartCycle() bool {
 	v1api := promv1.NewAPI(client)
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
-	result, warnings, err := v1api.Query(ctx, "cluster_autoscaler_last_activity{activity='scaleUp'}", time.Now())
+	// scaleDown metric will stops updating and report the last time it's updated when the cluster is scaling up
```

Comment:
I find this comment a little confusing - we are using the scaleDown dimension, but it is updated when the cluster is scaling up?

Reply:
Ah yes, because last_activity scaleDown and scaleUp is updated when the controller is performing the activity of checking if it needs to scaleDown or scaleUp, not when it is scaling down and up. So the end effect is that both metric continuously gets updated every cycle if no event is happening. The scaleDown metric however does not get updated when the the cluster is scaling up since there is no need to perform any check for if scaleDown is needed when a scale up event is happening. 

The metric behavior is documented on the `cluster-autoscaler` repo https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/proposals/metrics.md#cluster-autoscaler-execution

Reply:
👍 Thanks for the explanation!